python/BioSimSpace/Align/_align.py

```diff
@@ -308,13 +308,39 @@ def generateNetwork(molecules, names=None, work_dir=None, plot_network=False,
     else:
         if names is not None:
             for x, (molecule, name) in enumerate(zip(molecules, names)):
-                _IO.saveMolecules(work_dir + f"/inputs/{x:03d}_{name}",
-                    molecule, "mol2", property_map=property_map)
-                links_names[name] = f"{x:03d}_{name}.mol2"
+                # If the molecule came from an SDF file, then use
+                # that as the format as it's generally more reliable.
+                is_sdf = False
+                if molecule._sire_object.hasProperty("fileformat"):
+                    if "SDF" in molecule._sire_object.property("fileformat").value():
+                        is_sdf = True
+                        _IO.saveMolecules(work_dir + f"/inputs/{x:03d}_{name}",
+                            molecule, "sdf", property_map=property_map)
+                        links_names[name] = f"{x:03d}_{name}.sdf"
+                    else:
+                        _IO.saveMolecules(work_dir + f"/inputs/{x:03d}_{name}",
+                            molecule, "mol2", property_map=property_map)
+                        links_names[name] = f"{x:03d}_{name}.mol2"
+                else:
+                    _IO.saveMolecules(work_dir + f"/inputs/{x:03d}_{name}",
+                        molecule,  "mol2", property_map=property_map)
+                    links_names[name] = f"{x:03d}_{name}.mol2"
         else:
             for x, molecule in enumerate(molecules):
-                _IO.saveMolecules(work_dir + f"/inputs/{x:03d}",
-                    molecule, "mol2", property_map=property_map)
+                # If the molecule came from an SDF file, then use
+                # that as the format as it's generally more reliable.
+                is_sdf = False
+                if molecule._sire_object.hasProperty("fileformat"):
+                    if "SDF" in molecule._sire_object.property("fileformat").value():
```

Comment:
Should we unit test this part?

Reply:
Can do, although I'll have to think about the best way of doing it. I've also realised that I need to make sure it'll work with mixtures of file extensions. (I'm not sure if this is even possible with LOMAP.)